GitUI/UserControls/RevisionGrid.cs

```diff
@@ -1083,10 +1080,8 @@ public void ForceRefreshRevisions()
                 //Only check for tracked files. This usually makes more sense and it performs a lot
                 //better then checking for untracked files.
                 // TODO: Check FiltredFileName
-                Task<bool> unstagedChanges =
-                    Task.Factory.StartNew(() => Module.GetUnstagedFiles().Any());
-                Task<bool> stagedChanges =
-                    Task.Factory.StartNew(() => Module.GetStagedFiles().Any());
+                Task<IList<GitItemStatus>> changedFiles =
+                    Task.Factory.StartNew(() => Module.GetAllChangedFiles());
```

Comment:
since you're modifying it, this needs to change from `Task.Factory.StartNew` to `Task.Run`

* https://blog.stephencleary.com/2013/08/startnew-is-dangerous.html
* https://blogs.msdn.microsoft.com/pfxteam/2011/10/24/task-run-vs-task-factory-startnew/

Reply:
Will change